### PR TITLE
Na updating search messages

### PR DESCRIPTION
**Pull request by NidaAkkiswala**
Created the Search bar to search for messages within the chat.

### CODE REVIEW

File: frontend/src/app/chat/ChatMenu.jsx [hunk 1]
```diff
@@ -72,12 +72,12 @@ export function ChatMenu({ menuConfig, onMenuClick }) {
...
     const closeOption = () => setShowOption(null)
     const selectedOption = subMenuMap?.[showOption]
-    const { chat } = useContext(ChatContext)
+    const { chat, messages } = useContext(ChatContext)
     if (selectedOption) {
         const SubMenuComponent = selectedOption?.component || ComingSoon
         return (
             <SubMenuContainer name={selectedOption?.name} onClick={closeOption}>
-                <SubMenuComponent chat={chat} config={selectedOption} />
+                <SubMenuComponent chat={chat} config={selectedOption} messages={messages} />
```
[the remark this answers is not in the document]
Comment (vingkan) [reply frontend/src/app/chat/ChatMenu.jsx:80]: Nevermind, actually #295 modified the conversation starter props, not the chat menu. I saw you updated this branch, and it looks like it's up to date now!

File: frontend/src/app/chatmenu/SearchMessages.css [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    outline-color: rgba(0, 0, 0, 0);
+}
+
+.Search .Input {
+    --input-width: calc(80% - var(--send-width));
+    min-width: var(--input-width);
+    max-width: var(--input-width);
+    border-top-left-radius: var(--chat-composer-radius);
+    border-bottom-left-radius: var(--chat-composer-radius);
+    padding: 1em;
```
Comment (vingkan) [inline frontend/src/app/chatmenu/SearchMessages.css:24]: Nice use of CSS variables!

File: frontend/src/app/chatmenu/ChatMenuPage.tsx [hunk 3]
```diff
@@ -11,7 +11,16 @@ export interface ChatMenuConfig {
     options: Array<ChatMenuPageConfig>
 }
 
+export interface MessagesData {
+    key: string
+    type: string
+    timestamp: number
+    message: string
+    from: string
+}
```
Comment (vingkan) [inline frontend/src/app/chatmenu/ChatMenuPage.tsx:20]: Thanks for adding this! I didn't realize we didn't have a Message type definition in the frontend anywhere.

In the future, we might move this to a common location so that other components can use it, but let's keep it here for now.

File: frontend/src/app/chatmenu/SearchMessages.stories.tsx [hunk 4]
```diff
@@ -9,8 +10,34 @@ export default {
...
+        message: 'Not much, how about you?',
+    },
+]
+
+export const EmptySearch = Template.bind({})
+EmptySearch.args = { ...baseChatData }
 
-// TODO: Add more cases to test your component
+export const FoundSearch = Template.bind({})
+FoundSearch.args = { ...baseChatData, messages: [...shortMessages] }
```
Comment (vingkan) [inline frontend/src/app/chatmenu/SearchMessages.stories.tsx:43]: Nice job setting up these stories using the mock users and destructuring!

You don't have to do this, because this feature is working and the stories provide a way to test the search feature, but another way to make this code even more testable would be to separate the search bar and the filtered messages into their own components. Then you can pass in a search term to the filtered messages component to test whether it filters correctly. This allows you to decouple testing the filtering from the input, and test many edge cases quickly (lowercase, trimming, partial matches, etc.).

This is a common React pattern of splitting a feature into smaller components and "lifting state up" so that each component is not tightly coupled to another. If you are interested, you can read more about this in the official React docs: [Lifting State Up](https://reactjs.org/tutorial/tutorial.html#lifting-state-up).

File: frontend/src/app/chatmenu/SearchMessages.tsx [hunk 5]
```diff
@@ -1,19 +1,109 @@
...
+        ctrlKey: any
+        shiftKey: any
+        preventDefault: () => void
+    }) => {
+        const hitEnter = e.keyCode === 13 // Assigns hitEnter to the Enter key
+        const holdingKey = e.ctrlKey || e.shiftKey
+        if (hitEnter && !holdingKey) {
+            // Search upon hitting the enter key
+            e.preventDefault()
+            doSearch()
```
Comment (vingkan) [inline frontend/src/app/chatmenu/SearchMessages.tsx:77]: Nice job setting up this event handler and allowing the user to type multiple lines without submitting the search!

**Review by vingkan — APPROVED** [review]
Great work @NidaAkkiswala! This looks good to me to merge, all my comments are optional.

If you want to make any changes I suggested, you could merge this pull request, then open a new pull request to make the revisions, that way it will be easier to review the changes.

I approved the PR, so you can merge when you are ready! If @sprihajha wants you to make other changes, those could also be done in a follow-up PR.

File: frontend/src/app/chatmenu/SearchMessages.tsx [hunk 6]
```diff
@@ -1,19 +1,109 @@
...
+                const searchString: string = value ?? ''
+
+                return (
+                    text.message.toLowerCase().includes(searchString.toLowerCase()) && // Changes the message to lowercase
+                    searchString !== '' && // Checks for whitespaces
+                    searchString.trim()
+                )
+            })
+        )
+        setValue('')
```
[the remark this answers is not in the document]
Comment (sprihajha) [reply frontend/src/app/chatmenu/SearchMessages.tsx:63]: Thanks for pointing this out @vingkan 
We did discuss earlier that this could be part of the revisions. We could retain the search string and add `cross` icon to help user clear it prompty, similar to how Google search does it.

File: frontend/src/app/chatmenu/SearchMessages.stories.tsx (hunk 4 above)
Comment (sprihajha) [reply frontend/src/app/chatmenu/SearchMessages.stories.tsx:43]: Great point here! @vingkan 
I'll set this up as an upcoming task and pair program together to separate out these components and add search string based test cases.

**Review by sprihajha — APPROVED** [review]
Great job @NidaAkkiswala 🎉 
Looks good to me! 🚀

I'll set up all the suggestions @vingkan mentioned as tasks, and let's tackle that in a separate PR. Alongside that, we can also add the functionality to highlight the search string in the filtered messages.
